### PR TITLE
Update `test-infra` dependency

### PR DESCRIPTION
**Pull request by smg247**
Bumping `test-infra`. Quite a few breaking changes here, but they were fairly straightforward to port. A noteworthy change to the secret censor now results in 'X' instead of '*' in the output.

### CODE REVIEW

**Comment by smg247** [comment]
https://github.com/openshift/release/pull/28126 will take care of the `breaking-changes` test failure

File: pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml [hunk 1]
```diff
@@ -103,6 +103,7 @@
       image: sidecar
       name: sidecar
       resources: {}
+      terminationMessagePolicy: FallbackToLogsOnError
```
Comment (hongkailiu) [inline pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml:106]: I bet this is the source of breaking

Comment (hongkailiu) [reply pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml:106]: I lost the bet.

**Comment by hongkailiu** [comment]
LGTM.

I guess we need to fix the breaking?

File: pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml (hunk 1 above)
Comment (smg247) [reply pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml:106]: It's actually the new config I am specifically trying to get by bumping this that is causing the breaking changes.

**Comment by smg247** [comment]
> LGTM.
> 
> I guess we need to fix the breaking?

https://github.com/openshift/release/pull/28126 will do it

**Comment by hongkailiu** [comment]
/lgtm
/hold

Cancel whenever you are ready.

**Pull request merged**